// ocm-sdk/resources/index.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Referencedata = exports.Profile = exports.Poi = exports.OpenAPI = exports.Mediaitem = exports.Comment = void 0;
var comment_1 = require("./comment.js");
Object.defineProperty(exports, "Comment", { enumerable: true, get: function () { return comment_1.Comment; } });
var mediaitem_1 = require("./mediaitem.js");
Object.defineProperty(exports, "Mediaitem", { enumerable: true, get: function () { return mediaitem_1.Mediaitem; } });
var openapi_1 = require("./openapi.js");
Object.defineProperty(exports, "OpenAPI", { enumerable: true, get: function () { return openapi_1.OpenAPI; } });
var poi_1 = require("./poi.js");
Object.defineProperty(exports, "Poi", { enumerable: true, get: function () { return poi_1.Poi; } });
var profile_1 = require("./profile.js");
Object.defineProperty(exports, "Profile", { enumerable: true, get: function () { return profile_1.Profile; } });
var referencedata_1 = require("./referencedata.js");
Object.defineProperty(exports, "Referencedata", { enumerable: true, get: function () { return referencedata_1.Referencedata; } });
