// ocm-mcp/tools/referencedata/retrieve-referencedata.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("ocm-mcp/tools/types");
exports.metadata = {
    resource: 'referencedata',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/referencedata',
    operationId: 'get-referencedata',
};
exports.tool = {
    name: 'retrieve_referencedata',
    description: 'Returns the core reference data used for looking up IDs such as Connection Types, Operators, Countries etc. \n\nThis information is useful for UIs such as editing systems or for fetching results in the lighter non-verbose mode, then hydrating POI results back into complex objects.',
    inputSchema: {
        type: 'object',
        properties: {
            countryid: {
                type: 'array',
                description: 'Optional filter on countryid, exact match on a given numeric country id (comma separated list)',
                items: {
                    type: 'object',
                },
            },
        },
        required: [],
    },
    annotations: {
        readOnlyHint: true,
    },
};
const handler = async (client, args) => {
    const body = args;
    return (0, types_1.asTextContentResult)(await client.referencedata.retrieve(body));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
