// ocm-mcp/tools/referencedata/retrieve-referencedata.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { asTextContentResult } from 'ocm-mcp/tools/types';
export const metadata = {
    resource: 'referencedata',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/referencedata',
    operationId: 'get-referencedata',
};
export const tool = {
    name: 'retrieve_referencedata',
    description: 'Returns the core reference data used for looking up IDs such as Connection Types, Operators, Countries etc. \n\nThis information is useful for UIs such as editing systems or for fetching results in the lighter non-verbose mode, then hydrating POI results back into complex objects.',
    inputSchema: {
        type: 'object',
        properties: {
            countryid: {
                type: 'array',
                description: 'Optional filter on countryid, exact match on a given numeric country id (comma separated list)',
                items: {
                    type: 'object',
                },
            },
        },
        required: [],
    },
    annotations: {
        readOnlyHint: true,
    },
};
export const handler = async (client, args) => {
    const body = args;
    return asTextContentResult(await client.referencedata.retrieve(body));
};
export default { metadata, tool, handler };
