// ocm-mcp/tools/comment/submit-comment.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const filtering_1 = require("ocm-mcp/filtering");
const types_1 = require("ocm-mcp/tools/types");
exports.metadata = {
    resource: 'comment',
    operation: 'write',
    tags: [],
    httpMethod: 'post',
    httpPath: '/comment',
    operationId: 'post-comment',
};
exports.tool = {
    name: 'submit_comment',
    description: "When using this tool, always use the `jq_filter` parameter to reduce the response size and improve performance.\n\nOnly omit if you're sure you don't need the data.\n\nSubmit a user comment or checkin for a specific charging location\n\n# Response Schema\n```json\n{\n  type: 'object',\n  properties: {\n    description: {\n      type: 'string'\n    },\n    status: {\n      type: 'string'\n    }\n  },\n  required: [    'description',\n    'status'\n  ]\n}\n```",
    inputSchema: {
        type: 'object',
        properties: {
            chargePointID: {
                type: 'integer',
                description: 'This must be a valid POI ID',
            },
            checkinStatusTypeID: {
                type: 'integer',
                description: 'Optional valid CheckStatusTypeID to indicate overall catgeory and success/failure to use equipment e.g. 10 = Charged Successfully.',
            },
            comment: {
                type: 'string',
                description: 'This is an optional comment to describe the charging experience, may include guidance for future users.',
            },
            commentTypeID: {
                type: 'integer',
                description: 'This must be a valid Comment Type ID as per UserCommentTypes found in Core Reference Data. If left as null then General Comment will be used.',
            },
            rating: {
                type: 'integer',
                description: 'Optional integer rating between 1 = Worst, 5 = Best.',
            },
            relatedURL: {
                type: 'string',
                description: 'Optional website URL for related information',
            },
            userName: {
                type: 'string',
                description: 'This is an optional name to associate with the submission, for authenticated users their profile username is used.',
            },
            jq_filter: {
                type: 'string',
                title: 'jq Filter',
                description: 'A jq filter to apply to the response to include certain fields. Consult the output schema in the tool description to see the fields that are available.\n\nFor example: to include only the `name` field in every object of a results array, you can provide ".results[].name".\n\nFor more information, see the [jq documentation](https://jqlang.org/manual/).',
            },
        },
        required: ['chargePointID'],
    },
    annotations: {},
};
const handler = async (client, args) => {
    const { jq_filter, ...body } = args;
    return (0, types_1.asTextContentResult)(await (0, filtering_1.maybeFilter)(jq_filter, await client.comment.submit(body)));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
