// ocm-mcp/tools/openapi/retrieve-openapi.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("ocm-mcp/tools/types");
exports.metadata = {
    resource: 'openapi',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/openapi',
    operationId: 'get-openapi',
};
exports.tool = {
    name: 'retrieve_openapi',
    description: 'Retrieve the current OpenAPI format (YAML) definition for this API. This is useful for documentation tools, mocking, testing and client generation.',
    inputSchema: {
        type: 'object',
        properties: {},
        required: [],
    },
    annotations: {
        readOnlyHint: true,
    },
};
const handler = async (client, args) => {
    return (0, types_1.asTextContentResult)((await client.openAPI.retrieve()));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
